mailer/engine.py

```diff
@@ -152,19 +152,35 @@ def send_all():
                     logging.warning("message discarded due to failure in converting from DB. Added on '%s' with priority '%s'" % (message.when_added, message.priority))  # noqa
                 message.delete()
 
-            except (socket_error, smtplib.SMTPSenderRefused,
-                    smtplib.SMTPRecipientsRefused,
+            except (socket_error,
```

Comment:
Hmm... in Python 3.4, the base `smtplib.SMTPException` became a subclass of `OSError`, and `socket_error` *is* `OSError`. So this line catches *every* SMTPException -- even ones we'd like to fall through to the non-retryable case later.